syntax = "proto3";

package eleservice;

// 数据请求消息
message DataRequest {
  int32 id = 1;
  string name = 2;
  string value = 3;
  map<string, string> metadata = 4;
}

// 数据响应消息
message DataResponse {
  bool success = 1;
  string message = 2;
  DataItem data = 3;
  int32 total = 4;
}

// 数据项
message DataItem {
  int32 id = 1;
  string name = 2;
  string value = 3;
  map<string, string> metadata = 4;
  string created_at = 5;
  string updated_at = 6;
}

// 列表请求
message ListRequest {
  int32 page = 1;
  int32 page_size = 2;
  string search = 3;
}

// 列表响应
message ListResponse {
  bool success = 1;
  string message = 2;
  repeated DataItem items = 3;
  int32 total = 4;
}

// 空请求
message Empty {}

// 健康检查响应
message HealthResponse {
  string status = 1;
  string timestamp = 2;
  map<string, string> services = 3;
}

// 数据服务
service DataService {
  // 获取单个数据
  rpc GetData(DataRequest) returns (DataResponse);
  
  // 创建数据
  rpc CreateData(DataRequest) returns (DataResponse);
  
  // 更新数据
  rpc UpdateData(DataRequest) returns (DataResponse);
  
  // 删除数据
  rpc DeleteData(DataRequest) returns (DataResponse);
  
  // 获取数据列表
  rpc ListData(ListRequest) returns (ListResponse);
  
  // 流式获取数据
  rpc StreamData(ListRequest) returns (stream DataResponse);
  
  // 流式处理数据
  rpc ProcessData(stream DataRequest) returns (DataResponse);
  
  // 健康检查
  rpc HealthCheck(Empty) returns (HealthResponse);
} 